core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java

```diff
@@ -481,12 +487,49 @@ public static void deleteCancerStudy(String cancerStudyStableId) throws DaoExcep
             JdbcUtil.closeAll(DaoCancerStudy.class, con, pstmt, rs);
         }
     }
+    
+    /**
+     * Calls deleteCancerStudyByCascade() if cancer study exists by stable id.
+     * @param cancerStudyStableId
+     * @throws DaoException 
+     */
+    public static void deleteCancerStudyByCascade(String cancerStudyStableId) throws DaoException {
+        CancerStudy study = getCancerStudyByStableId(cancerStudyStableId);
+        if (study != null) {
+            deleteCancerStudyByCascade(study.getInternalId());
+        }
+    }
 
     /**
+     * Deletes a cancer study by internal id from db with foreign key constraints enforced.
+     * @param internalCancerStudyId
+     * @throws DaoException 
+     */
+    static void deleteCancerStudyByCascade(int internalCancerStudyId) throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoCancerStudy.class);
+            pstmt = con.prepareStatement("DELETE FROM cancer_study WHERE cancer_study_id = ?");
+            pstmt.setInt(1, internalCancerStudyId);
+            pstmt.executeUpdate();
```

Comment:
I think executeUpdate() is going to return the number of records altered. If this returns zero, then the deletion did not happen, and we probably should throw a DaoException to let the caller know that the call failed.
("Could not find cancer_study_id in table cancer_study")
An interesting question is --- when it succeeds, I wonder if the returned value is 1 (for the one record deleted from cancer_study) or a bigger number (for all the cascaded deletes that happen across all tables). I have no idea what the behavior is, but would guess we will get the bigger number.